src/tests/Microsoft.VisualStudio.Composition.Tests/Reflection/TypeRefTests.cs

```diff
@@ -47,6 +47,28 @@ public void EqualsChecksMvidEquality()
             this.TestAssemblyNameEqualityNotEqual(assemblyName, assemblyName, @"C:\MyAssembly.dll", @"C:\MyAssembly.dll", guidV1, guidV2);
         }
 
+        [Fact]
+        public void ThrowArgumentExceptionWhenArgIsNull()
+        {
+            var testGuid = new Guid("00000000-0000-0000-0000-000000000001");
+            var loadSystemAssembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(assembly => assembly.FullName.Contains("mscorlib"));
+            Assert.NotNull(loadSystemAssembly);
+
+            var loadedAssemblyName = new AssemblyName(loadSystemAssembly.FullName)
+            {
+                CodeBase = loadSystemAssembly.CodeBase
+            };
+            var assemblyIdentity = new StrongAssemblyIdentity(loadedAssemblyName, testGuid);
+
+            var typeRefNullableArgument2 = TypeRef.Get(TestUtilities.Resolver, assemblyIdentity, 0x02000001, "Fake assembly", TypeRefFlags.None, 0, ImmutableArray<TypeRef>.Empty, false, ImmutableArray<TypeRef>.Empty, null);
+
+            var typeRef = TypeRef.Get(TestUtilities.Resolver, assemblyIdentity, 0x02000001, typeof(Dictionary<,>).FullName, TypeRefFlags.None, 0, new[] { typeRefNullableArgument2 }.ToImmutableArray(), false, new[] { typeRefNullableArgument2 }.ToImmutableArray(), null);
+
+            var actualException = Assert.Throws<TypeLoadException>(() => typeRef.Resolve());
+            Assert.NotNull(actualException);
```

Comment:
tip: you don't need to assert that `actualException` is not null. `Assert.Throws` guarantees to return a non-null result.

Reply:
Thank you for providing feedback, I have updated the test case.